engines/hadesch/video.cpp

```diff
@@ -0,0 +1,1040 @@
+		if (it->name == name) {
+			return true;
+		}
+	}
+	return false;
+}
+
+bool VideoRoom::isAnimationFinished(const LayerId &name, int time) {
+	for (Common::SortedArray<Layer>::iterator it = _layers.begin(); it != _layers.end(); it++) {
+		if (it->name == name) {
+			return it->renderable->isAnimationFinished(time);
+		}
+	}
+
+	return true;
+}
+
+void VideoRoom::addLayer(Renderable *rend, const LayerId &name, int zValue,
+			 bool isEnabled, Common::Point offset) {
+	Layer l;
+	l.renderable = Common::SharedPtr<Renderable>(rend);
+	l.name = name;
+	l.isEnabled = isEnabled;
+	l.offset = offset;
+	l.zValue = zValue;
+	l.parallax = 0;
+	l.colorScale = 0x100;
+	l.scale = 100;
+	l.genCounter = _layerGenCounter++;
+	_layers.insert(l);
+}
+
+void VideoRoom::computeHotZone(int time, Common::Point mousePos) {
+	bool wasEdge = _leftEdge || _rightEdge;
+	_leftEdge = false;
+	_rightEdge = false;
+
+	Common::SharedPtr<HeroBelt> belt = g_vm->getHeroBelt();
+	
+	if (_heroBeltEnabled)
+		belt->computeHotZone(time, mousePos, _mouseEnabled);
+	if (_heroBeltEnabled && belt->isOverHeroBelt()) {
+		_hotZone = -1;
+		return;
+	}
+	if (!_mouseEnabled) {
+		_hotZone = -1;
+		return;
+	}
+	Common::Point canvasPos = mousePos + Common::Point(_pan, 0);
+	int wasHotZone = _hotZone;
+	_hotZone = _hotZones.pointToIndex(canvasPos);
+	if (_hotZone >= 0) {
+		if (wasHotZone < 0) {
+			_startHotTime = time;
+		}
+		return;
+	}
+	_leftEdge = (mousePos.x < 20) && _pan > 0;
+	_rightEdge = (mousePos.x > 620) && _pan < 640;
+	if ((!wasEdge && (_leftEdge || _rightEdge)))
+		_edgeStartTime = time;
+	_hotZone = -1;
+}
+
+void VideoRoom::setHotZoneOffset(const Common::String &name, Common::Point offset) {
+	_hotZones.setHotZoneOffset(name, offset);
+}
+
+Common::String VideoRoom::mapClick(Common::Point mousePos) {
+	if (!_mouseEnabled) {
+		return "";
+	}
+	Common::Point canvasPos = mousePos + Common::Point(_pan, 0) + _viewportOffset;
+	return _hotZones.pointToName(canvasPos);
+}
+
+Common::String VideoRoom::getHotZone() {
+	return _hotZones.indexToName(_hotZone);
+}
+
+int VideoRoom::getCursorAnimationFrame(int time) {
+	if (_hotZone < 0)
+		return -1;
+	if (_hotZones.indexToICSH(_hotZone) != 0)
+		return -1;
+	return (time - _startHotTime) / kDefaultSpeed;
+}
+
+bool VideoRoom::isVideoPlaying() {
+	return !!_videoDecoder;
+}
+
+void VideoRoom::fadeOut(int ms, const EventHandlerWrapper &callback) {
+	_finalFadeCallback = callback;
+	_finalFadeSpeed = -(kDefaultSpeed * 0x100) / ms;
+}
+
+void VideoRoom::nextFrame(Common::SharedPtr<GfxContext> context, int time, bool stopVideo) {
+	context->clear();
+
+	Common::SharedPtr<HeroBelt> belt = g_vm->getHeroBelt();
+
+	if (!_mouseEnabled) {
+		_cursor = (time / 200) % 10 + 3;
+	} else if (_heroBeltEnabled && belt->isOverHeroBelt()) {
+		_cursor = belt->getCursor(time);
+	} else if (g_vm->getPersistent()->_currentRoomId == kMonsterPuzzle)
+		_cursor = 17;
+	else if (_hotZone >= 0) {
```

Comment:
I would put curly brackets here just for consistency

Reply:
Done